src/bitmap/backend/atomic_bitmap.rs

```diff
@@ -76,8 +84,37 @@ impl AtomicBitmap {
                 // Attempts to set bits beyond the end of the bitmap are simply ignored.
                 break;
             }
-            self.map[n >> 6].fetch_or(1 << (n & 63), Ordering::SeqCst);
+            if reset {
+                self.map[n >> 6].fetch_and(!(1 << (n & 63)), Ordering::SeqCst);
+            } else {
+                self.map[n >> 6].fetch_or(1 << (n & 63), Ordering::SeqCst);
+            }
+        }
+    }
+
+    /// Reset a range of `len` bytes starting at `start_addr`. The first bit set in the bitmap
+    /// is for the page corresponding to `start_addr`, and the last bit that we set corresponds
+    /// to address `start_addr + len - 1`.
+    pub fn reset_addr_range(&self, start_addr: usize, len: usize) {
+        self.set_reset_addr_range(start_addr, len, true);
```

Comment:
Just a nitpick, but it's a bit counter-intuitive in my opinion that the third argument being set to `true` means that the bits are set to zero, I would expect the opposite. Otherwise LGTM.

Reply:
@00xc  Fixed now